src/blame.c

```diff
@@ -455,12 +451,9 @@ blame_request(struct view *view, enum request request, struct line *line)
 			if (diff->pipe)
 				string_copy_rev(diff->ref, NULL_ID);
 		} else {
-			if (filename_from_blame) {
-				free(filename_from_blame);
-			}
-			filename_from_blame = strdup(blame->commit->filename);
-			lineno_from_blame = blame -> lineno;
-			open_diff_view(view, flags | OPEN_RELOAD | OPEN_BLAMED_LINE);
+			string_ncopy(view->env->file, blame->commit->filename, strlen(blame->commit->filename));
```

Comment:
Is it impossible for strlen(blame->commit->filename) to be  more than SIZEOF_STR long?

Reply:
Maybe not, but the diff view already relies on it.